action-server/covidflow/actions/question_answering_form.py

```diff
@@ -67,128 +63,109 @@
-        dispatcher.utter_message(template="utter_qa_disclaimer")
-
-        random_qa_samples = (
-            _get_fixed_questions_samples()
-            if _must_stub_result(tracker)
-            else _get_random_question_samples(domain)
-        )
-
-        if len(random_qa_samples) > 0:
-            dispatcher.utter_message(
-                template="utter_qa_sample",
-                sample_questions="\n".join(random_qa_samples),
+            random_qa_samples = (
+                _get_fixed_questions_samples()
+                if _must_stub_result(tracker)
+                else _get_random_question_samples(domain)
             )
 
-        return await super()._activate_if_required(dispatcher, tracker, domain) + [
-            SlotSet(SKIP_QA_INTRO_SLOT, True)
-        ]
+            if len(random_qa_samples) > 0:
+                dispatcher.utter_message(
+                    template="utter_qa_sample",
+                    sample_questions="\n".join(random_qa_samples),
+                )
+            events = [SlotSet(SKIP_QA_INTRO_SLOT, True)]
 
-    @staticmethod
-    def required_slots(tracker: Tracker) -> List[Text]:
-        status = tracker.get_slot(STATUS_SLOT)
-        if status == QuestionAnsweringStatus.SUCCESS:
-            return [QUESTION_SLOT, FEEDBACK_SLOT]
+        dispatcher.utter_message(template="utter_ask_active_question")
+        return events
 
-        return [QUESTION_SLOT]
 
-    def slot_mappings(self) -> Dict[Text, Union[Dict, List[Dict]]]:
-        return {
-            QUESTION_SLOT: self.from_text(),
-            FEEDBACK_SLOT: yes_no_nlu_mapping(self),
-        }
+class ValidateQuestionAnsweringForm(Action):
+    def name(self) -> Text:
+        return VALIDATE_ACTION_NAME
+
+    async def run(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    ) -> List[EventType]:
+        bind_logger(tracker)
+
+        extracted_slots: Dict[Text, Any] = _form_slots_to_validate(tracker)
+
+        validation_events: List[EventType] = []
+
+        for slot_name, slot_value in extracted_slots.items():
+            slot_events = [SlotSet(slot_name, slot_value)]
+
+            if slot_name == FEEDBACK_SLOT:
+                if slot_value is False:
+                    dispatcher.utter_message(template="utter_feedback_false")
+                elif not isinstance(slot_value, bool):
+                    slot_events = [SlotSet(FEEDBACK_SLOT, FEEDBACK_NOT_GIVEN)]
+            elif slot_name == QUESTION_SLOT:
+                result = (
+                    _get_stub_qa_result(tracker)
+                    if _must_stub_result(tracker)
+                    else await _fetch_qa(slot_value, tracker)
+                )
+
+                slot_events += [SlotSet(STATUS_SLOT, result.status)]
+
+                if (
+                    result.status == QuestionAnsweringStatus.SUCCESS and result.answers
+                ):  # not sure about this. Shouldn't happen
```

Comment:
Pourquoi ça ne devrait pas arriver?

Reply:
Oups. Ça devrait arriver tout le temps en fait. Le `and` en gros je sais pas pourquoi il est là, si la liste est vide alors qu'on a un succès c'est vraiiiiiiiment louche.

Reply:
(J'ai enlevé le check des réponses. Si y'a rien, on est dans la merde anyway...)

Reply:
Ok, je suis soulagé de savoir que le happy path va arriver :)